### PR TITLE
SQL store : refactor "_get_run" method.

### PR DESCRIPTION
**Pull request by mparkhe**
- callers handle lifecycle_stage
- add lifecycle_stage checks to logging API
- unit tests

### CODE REVIEW

**Comment by mparkhe** [comment]
@aarondav : fix based on your suggestion in PR #844 [discussion_r251086060](https://github.com/mlflow/mlflow/pull/844#discussion_r251086060)

File: mlflow/store/sqlalchemy_store.py [hunk 1]
```diff
@@ -152,8 +150,19 @@ def _get_run(self, run_uuid, view_type):
 
         return runs[0]
 
+    def _check_run_is_active(self, run):
+        if run.lifecycle_stage != LifecycleStage.ACTIVE:
+            raise MlflowException("The run {} must be in 'active' lifecycle_stage."
```
Comment (aarondav) [inline mlflow/store/sqlalchemy_store.py]: Would it be meaningful if we said what state it currently is?

File: tests/store/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -498,23 +498,72 @@ def test_restore_experiment(self):
...
+
+        with self.assertRaises(MlflowException) as e:
+            self.store.log_metric(run_uuid, entities.Metric("m1345", 1.0, 123))
+        self.assertIn("must be in 'active' lifecycle_stage", e.exception.message)
+
+        with self.assertRaises(MlflowException) as e:
+            self.store.set_tag(run_uuid, entities.RunTag("t1345", "tv1"))
+        self.assertIn("must be in 'active' lifecycle_stage", e.exception.message)
+
+        # restore this run and try again
```
Comment (aarondav) [inline tests/store/test_sqlalchemy_store.py]: nice, like the restore part

**Comment by aarondav** [comment]
Looks good, just one question regarding the message

**Pull request merged**